### ISSUE TITLE
Split  rsyslog__capabilities to inputs and outputs to support multiple outputs

### ISSUE THREAD

**Issue by sradco** [linux-system-roles/logging#17]
Split  rsyslog__capabilities to  outputs so we can configure each input to each output/s.

Inputs - viaq, viaq-k8s, rsyslog example, ovirt-host, ovirt-engine, custom
Outputs -  P1 - Viaq, P1 - Local (File/Journal), P1 - Remote Rsyslog, P2 - Elasticsearch, P2 - Fluentd Forward, P2 - Message Queue (kafka, amqp)

@richm @nhosoi @pcahyna @nkinder

**Comment by nhosoi** [comment]
@sradco, are you suggesting to have rsyslog_input_capabilities and rsyslog_output_capabilities instead of just one rsyslog_capabilities?  Thanks.

**Comment by sradco** [comment]
We need to work in stages to be able to send logs to multiple outputs.
Ats first step we need to separate the capabilities to the once responsible for setting up the logs collection based on the group like we have now: viaq, viaq-k8s, rsyslog example and default conf.
And I'll need to add the oVirt logs.

We also need to have a separate parameter that deploys the outputs. 
At first we can keep as is in terms of having viaq output - general and ops elasticsearch and I'll add the collection of ovirt logs that will send data to it as well. 
Also we have the remote rsyslog which I'm not sure if only supported for rsyslog example or for openshift and container logs and default conf as well.


@nhosoi @nhosoi @pcahyna @nkinder Please let me know if you agree with this.

**Comment by nhosoi** [comment]
> Ats first step we need to separate the capabilities to the once responsible for setting up the logs collection based on the group like we have now: viaq, viaq-k8s, rsyslog example and default conf.

Currently, rsyslog_viaq_rules corresponding to rsyslog_viaq variable defines these config files.  Do you think these are shared by oVirt?  I.e., a subset or some are in common but not others or a superset?  Unless it is a superset, I'd think you'd better have rsyslog_ovirt_rules to define the oVirt config file set?  Then, if you need further control in rsyslog_ovirt (e.g., I introduced viaq-k8s for configuring mmk8s or not depending on the deployed environment), you could define them?
```
  - '{{ rsyslog_conf_global_options }}'
  - '{{ rsyslog_conf_local_modules }}'
  - '{{ rsyslog_conf_network_modules }}'
  - '{{ rsyslog_conf_common_defaults }}'
  - '{{ rsyslog_conf_viaq_main_modules }}'
  - '{{ rsyslog_conf_viaq_mmk8s }}'
  - '{{ rsyslog_conf_viaq_elasticsearch }}'
  - '{{ rsyslog_conf_viaq_viaq_formatting }}'
  - '{{ rsyslog_rulebase_viaq_k8s_filename }}'
  - '{{ rsyslog_rulebase_viaq_parse_json }}'
  - '{{ rsyslog_rulebase_viaq_k8s_container_name }}'
  - '{{ rsyslog_json_viaq_normalize_level }}'
  - '{{ rsyslog_json_viaq_prio_to_level }}'
  - '{{ rsyslog_rulebase_viaq_crio }}'
```

**Comment by nhosoi** [comment]
Can we discuss about the extension for supporting RHV here?

First, I'd like to learn about the ovirt-host and ovirt-engine inputs.  How they pass the logs to rsyslog?  Could it be via a file, journald, a unix socket, or a tcp?

From the original comment by @sradco:
> Inputs - viaq, viaq-k8s, rsyslog example, ovirt-host, ovirt-engine, custom

- viaq takes input from journald.
- viaq-k8s reads from log files in /var/log/containers in addition to journald.
- rsyslog example receives inputs from a unix socket and tcp/udp.
- For the custom config files, we provide rsyslog_custom_config_files to deploy the given files to /etc/rsyslog.d.

Regarding the outputs,
> Outputs - P1 - Viaq, P1 - Local (File/Journal), P1 - Remote Rsyslog, P2 - Elasticsearch, P2 - Fluentd Forward, P2 - Message Queue (kafka, amqp)

- viaq (omelasticsearch) outputs viaq formatted logs to elasticsearch(es).
- Local (omfile, omjournal) is straightforward to add if the spec is given.
- Remote Rsyslog (omfwd) is straightforward; not secure; it requires imtcp/imudp.
- Remote Rsyslog (omrelp) is a reliable method, but it requires imrelp.
- Elasticsearch(?) - what's the difference from the viaq?  Non-viaq data model?
- Fluentd Forward - Do we want to send logs to fluentd by secure_forward?
- Message Queue, no much knowledge about this...  At least, rsyslog has the omkafka output plugin.
@richm,  could you correct me and/or add more info?  Thanks!

**Comment by richm** [comment]
For information about what ovirt currently does, take a look at the fluentd role: https://github.com/linux-system-roles/logging/tree/master/roles/fluentd

For example, here is one of the inputs: https://github.com/linux-system-roles/logging/blob/master/roles/fluentd/fluentd-inputs/templates/ovirt-vdsm-log.template
https://github.com/linux-system-roles/logging/tree/master/roles/fluentd/fluentd-inputs also mentions the engine log file, but I don't see a config for it.  So I think that's it - just the vdsm and engine log files for now, and probably journal in the future.

**Comment by nhosoi** [comment]
@sradco , may I ask why you needed to use multiline format here?
https://github.com/linux-system-roles/logging/blob/master/roles/fluentd/fluentd-inputs/templates/ovirt-vdsm-log.template#L7-L9
Could vdsm.log get log messages split somewhere in one line and folded?  It looks to me #L7-L9 does not handle such cases... (I might be missing something, of course...)

I'm wondering instead of #L7-L9, can we have just the format1 value for each log?
`format /(?<time>((\d{4})-(\d{2})-(\d{2}) (\d{2}):(\d{2}):(\d{2}),(\d{3})(.+?)))\s+(?<level>[^ ]+)\s+\((?<thread>.+?)\)\s+\[(?<module>.+?)\]\s+(?<message>.*)\s+\((?<module_lineno>.+?)\)/`

**Comment by nhosoi** [comment]
> Its not enough because some logs have a long log trace.

Hmmm, could you share some sample logs having the long log trace?  Thanks.

**Comment by sradco** [comment]
Fixed in #27
